rockset/resource_scheduled_lambda.go

```diff
@@ -0,0 +1,245 @@
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: rocksetNameValidator,
+			},
+			"apikey": {
+				Description: "The apikey to use when triggering execution of the associated query lambda.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// apikey is not returned for security reasons
+					return new == ""
+				},
+			},
+			"cron_string": {
+				Description: "The UNIX-formatted cron string for this scheduled query lambda.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"ql_name": {
+				Description: "The name of the QL to use for scheduled execution.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"tag": {
+				Description: "The QL tag to use for scheduled execution.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"version": {
+				Description: "The version of the QL to use for scheduled execution.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"total_times_to_execute": {
+				Description:  "The number of times to execute this scheduled query lambda. Once this scheduled query lambda has been executed this many times, it will no longer be executed.",
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntAtLeast(0),
+			},
+			"webhook_auth_header": {
+				Description: "The value to use as the authorization header when hitting the webhook.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// auth header is not returned for security reasons
+					return new == ""
+				},
+			},
+			"webhook_payload": {
+				Description: "The payload that should be sent to the webhook. JSON format.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"webhook_url": {
+				Description: "The URL of the webhook that should be triggered after this scheduled query lambda completes.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceScheduledLambdaCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	rc := meta.(*rockset.RockClient)
+	var diags diag.Diagnostics
+
+	workspace := d.Get("workspace").(string)
+	apikey := d.Get("apikey").(string)
+	cronString := d.Get("cron_string").(string)
+	qlName := d.Get("ql_name").(string)
+
+	options := getScheduledLambdaOptions(d)
+
+	scheduledLambda, err := rc.CreateScheduledLambda(ctx, workspace, apikey, cronString, qlName, options...)
+	if err != nil {
+		return DiagFromErr(err)
+	}
+
+	scheduledLambdaRrn := *scheduledLambda.Rrn
+	d.SetId(toID(workspace, scheduledLambdaRrn))
```

Comment:
the RRN should be unique, so I don't think you have to add the workspace, and we could even use the UUID part of the RRN, as this will always be cluster specific anyway

Reply:
ok, then we can keep it the way you have done it. it is just slightly cumbersome when you import an existing scheduled query lambda into terraform, but that doesn't happen very often